content/extend/gorma.md

```diff
@@ -6,3 +6,233 @@ title = "the gorma Plugin"
+			})
+			Field("title", func() {
+				Alias("proposal_title")
+			})
+			Field("created_at", gorma.Timestamp, func() {})
+			Field("updated_at", gorma.Timestamp, func() {})
+			Field("deleted_at", gorma.NullableTimestamp, func() {})
+			...
+		})
+```
+The "id" field is explicitly set here, but as mentioned above it doesn't need to be.  It also includes the `PrimaryKey()` definition in the field's DSL.   If you have compound primary keys, you can specify them by creating multiple field definitions in your Model with the `PrimaryKey()` tag.
+
+We've also specified an `Alias` for the "title" field, which tells `gorma` that the `title` field in the model will be stored in the `proposal_title` column in the database.  You can specify an Alias for both Model and Field definitions.  If you specify an Alias on a table, you're changing the name of the table in the database.
+
+
+## Expectations
+Gorma makes certain expectations, and you shouldn't deviate too far from them if you want all the automatic wiring to work well.
+* Your Model relationships need to match the implicit relationships you specify in your Media Types.  When you specify that a Media Type has an Attribute that is actually another Media Type you're specifying a relationship.  That relationship must be also reflected in your `gorma` DSL.
+* Naming:  Name your Models with `CamelCase`.  Name your Fields with `snake_case`.  `gorma` always tries to do the right things with naming and conversions, but keeping this standard will help with edge cases that aren't yet tested.
+
+
+## Limitations
+* Gorma expects that all of your keys (primary and foreign) will be `int` types.  In the future we may support non-integer primary keys.
+* Gorma is currently tested extensively against PostgreSQL.  It should work well against other databases supported by Gorm, and if it doesn't please file a bug.
+* The methods generated by `gorma` are basic case methods for data manipulation by primary key.  It's possible to add lookup functions in the future by other fields in the model.  Pull requests gratefully accepted.
+* Gorm ~ and any ORM, really ~ won't generate optimal SQL for every use case.  If you need to optimize a particular case, see the note under "Other Notes" below about adding additional methods to your Models, and hand-write your SQL.  `gorma` exposes the underlying database handle so you can drop down to Go's `db/sql` package and write anything you want in your custom methods.
+
+
+##  Other Notes
+* `gorma` generates your models into the `models` folder of your project's root.  It will overwrite these models every time you use `goagen`, but it will only overwrite files that it creates.  You can add extra files to this directory to specify additional functionality for your models as long as you use a file name that won't conflict with `gorma`'s naming convention:  `model.go` and `model_helper.go`.  We recommend something like `model_addons.go` to keep your extra code from being overwritten.
```

Comment:
We should add a note about the `--pkg` flag.  You can append `-- --pkg=genmodels` to your `goagen` command line to tell `gorma` to generate into a custom directory.

We may need a short section on how to execute `goagen` with `gorma`.